src/core/services/interimElement/interimElement.js

```diff
@@ -378,7 +398,7 @@ function InterimElementProvider() {
        * Note: interim elements are in "interim containers"
        */
       function destroy(target) {
-        var interim = !target ? stack.shift() : null;
+        var interim = !target ? stack.splice(hideQueue, 1)[0] : null;
```

Comment:
@EladBezalel When destroying, we don't care anymore about the animation (it won't run), so the item will be removed instantly.